features/246-custom-domain/index.md

```diff
@@ -0,0 +1,70 @@
+# Custom domain
+
+## Overview
+
+Allow developer to setup custom domain for their application, basically following features will be supported.
+
+- Allow developer to setup custom domain.
+- Let's Encrypt certificate will be provided by default, certificate will be renew automatically.
+- Developer can configure to use custom SSL certificates for their domain. They will need to update the certificates before it expires in this case.
+
+## Use Cases
+
+### Setup custom domain through skycli
+
+1. Add custom domain
+
+    ```sh
+    $ skycli domain add myapp.example.com --app=myapp
+    Added domain myapp.example.com successfully!
+
+    Add following DNS records in your DNS provider.
+
+    TYPE      HOST                     VALUE
+    TXT       _skygear.example.com     5636486ffc5a4dfebf4a13f480bd9a95
+    A         myapp.example.com          <ingress controller lb ip>
+
+    After updating DNS records, run `skycli domain verify myapp.example.com` to verify domain.
+    ```
+
+1. Developer update the DNS records through their DNS provider.
+
+1. Call Skygear cloud to verify the domain
+
+    ```sh
+    $ skycli domain verify myapp.example.com --app=myapp
+    Success! You can now access your app through myapp.example.com.
+    Your site may show a security certificate warning until the certificate has been provisioned.
+    ```
+
+1. (Optional) Setup custom certificate
+
+    ```sh
+    # Create tls secret
+    $ skycli secret create myapp-tls --type=tls --cert=path/to/tls.cert --key=path/to/tls.key --app=myapp
+    Success! Created secret myapp-tls
+
+    # Update domain using custom certificates
+    $ skycli domain update myapp.example.com --tls-secret=myapp-tls --app=myapp
```

Comment:
I suppose after running `skycli domain update`, I can run `skycli secret delete` on previous secret without impact on the operation? We may want to note that the developer need to rotate the certificate manually.

Do we have a way to complete erase a secret from the cluster? Our current implementation keeps the secrets forever.

Reply:
> I suppose after running skycli domain update, I can run skycli secret delete on previous secret without impact on the operation

yes, since we have not delete the k8s secret actually, so  `skycli secret delete` will not accept the existing domain (same as deployment).

> Do we have a way to complete erase a secret from the cluster? Our current implementation keeps the secrets forever.

We don't have this yet, let me open another issue to discuss this.

Reply:
I think at least we need a way for developer to erase certification private key for custom domain. If  developer wants to migrate the domain away, we need to ensure we don't keep the key materials longer than necessary.

For simplicity, we may have a cron job to remove unreferenced TLS secrets for now?

Reply:
I think no only the TLS secret, we can use the job to remove the unreferenced secrets for all types too :)